tests/tests.go

```diff
@@ -32,14 +33,22 @@ func StartServer(t testing.TB, cfg *serverconfig.Config) {
 	StartServerWithContext(t, cfg, serverCtx)
 }
 
-// StartServerWithContext starts a server with a specific ServerContext and waits until it is healthy.
+// StartServerWithContext starts a server in random ports and with a specific ServerContext and waits until it is healthy.
 // When the test ends, all resources are cleaned.
 func StartServerWithContext(t testing.TB, cfg *serverconfig.Config, serverCtx *run.ServerContext) {
 	container := storage.RunDatastoreTestContainer(t, cfg.Datastore.Engine)
 	cfg.Datastore.URI = container.GetConnectionURI(true)
 
 	ctx, cancel := context.WithCancel(context.Background())
 
+	httpPort, httpPortReleaser := testutils.TCPRandomPort()
+	cfg.HTTP.Addr = fmt.Sprintf("0.0.0.0:%d", httpPort)
+	grpcPort, grpcPortReleaser := testutils.TCPRandomPort()
+	cfg.GRPC.Addr = fmt.Sprintf("0.0.0.0:%d", grpcPort)
+
+	httpPortReleaser()
```

Comment:
Question: will it be better to have the httpPortReleaser() and grpcPortReleaser() be part of the t.Cleanup() to allow StaartServerWithContext to be ran as parallel in the future?

Reply:
No, we don't want to run these functions at the very end of the tests. The code flow should be like this:

1. `TCPRandomPort` finds a free port
2. We assign the port to the server config
3. We release the port because the server will use it.

If we run this at the end, the server will not be able to listen on that port.